### PR TITLE
(#548) Hero Cta Strip

### LINKED ISSUE #548: Build: Hero - G. CTA Strip
## As a user, I would like to see a CTA strip, so that I can quickly scan the calls to action. 

Build CTA Strip variation of hero component based on design specs/requirements documented in this ticket.

> **ESTIMATE** 3

### Styling: 
*Anatomy of a component*
- [ ] Use correct spacing (8px grid) size, emphasis, alignment

### Behaviors
*Summary of state changes and behaviors. AC's are more detailed.*
- If button wraps onto multiple lines, button will be centered vertically
- Buttons are fluid and equal width on all breakpoints
- Widescreen and desktop:
  - Buttons display in a row
  - margin-3 between buttons
- Tablet large, tablet, mobile large, mobile:
  - Buttons stack vertically 
  - margin-105 between stacked buttons
- If the button is an external link, it should get the external link icon (launch) instead of the >

### InVision: 
  * Widescreen (1400) mockup: https://publicis.invisionapp.com/console/MVP2-Landing-Pages-cl756hway0281013t70e36jn0/cld503pc8003m01d03cnd3ybu/play
  * Desktop (1024) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl7fbj88406h8012084nn6b5m/play
  * Tablet Large (880) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl7fbj88406h9012085ar4gy3/play
  * Tablet (640) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl7fbj88406ha012009l906i9/play
  * Mobile Large (480) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl7fbj88406hb01209rmf8b93/play
  * Mobile (320) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl7fbj88406hc0120hlg6frb7/play

### Acceptance Criteria for Behaviors 
*If a component changes between breakpoints, or things show/hide, that gets captured here* 

```gherkin
Scenario: User views a landing page with a CTA Strip on Widescreen and Desktop  
Given a user is viewing a landing page with a CTA Strip 
  And the browser width is desktop or widescreen
Then the CTA strip will appear 
  And the buttons will appear with the primary button style and vertically align
  And the buttons will have the same width
  And the buttons will have the same margin between them
  And on hover, the buttons will appear with the primary button hover style 
  And clicking on each button will take the user to the corresponding page 

Scenario: User views a landing page with a CTA Strip on Tablet Large, Tablet, Mobile Large or Mobile
Given a user is viewing a landing page with a CTA Strip 
  And the browser width is tablet-lg or less
Then the CTA strip will appear 
  And the buttons will appear with the primary button style and stack
  And the buttons will have the same width
  And the buttons will have the same margin between them
  And on hover, the buttons will appear with the primary button hover style 
  And clicking on each button will take the user to the corresponding page 
```

---

### Scenarios
- CTA button 1: Programs
- CTA button 2: Resources
- CTA button 3: Success Stories
- CTA button 1 long: NCI Alliance for Nanotechnology in Cancer
- CTA button 2 long: Nanotechnology Characterization Laboratory
- CTA button 3 long: Cancer Nanotechnology Plan
- Spanish CTA button 1: El NCI para la Nanotecnología en el Cáncer
- Spanish CTA button 2: Laboratorio de Caracterización de Nanotecnología
- Spanish CTA button 3: Plan de Nanotecnología del Cáncer

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*

- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Update Documentation Site (docs)
  - [ ] Create or update [INSERT DOCUMENTATION PAGE OR SECTION ON PAGE HERE] that includes:
    - Developer Documentation
      -  What SASS partial to import
      - (HTML Tab) What is the HTML?
      - (HTML Tab) (for vanilla js) What is the js library to import?
      - (React Tab) What is the react library to import?
      - (React Tab) What is the react markup?
    - **IA Provided** Guidance Documentation  
      - Includes when/how the component should be used
  - [ ] If the component is a global component like the footer, then it needs to be added to the documentation site as well as a real usage.
- [ ] Update NCIDS-CSS
  - **Insert list here of partials to be created** (There should be a sass partial available for all variants that are being created in this ticket)
- [ ] Update NCIDS-JS
  - **Insert requirements/list of components for JS**
- [ ] Update NCIDS-React
  - [ ] **Insert requirements/list of components for react**
    - (it can also be a separate ticket, should be linked here)

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #548

### CODE REVIEW

File: packages/ncids-css/scss/components/_all.scss [hunk 1]
```diff
@@ -38,3 +38,8 @@
 @import './usa-sidenav/nci-sidenav';
 @import './usa-site-alert/all';
 @import './usa-skipnav/usa-skipnav';
+@import 'uswds/dist/scss/components/step-indicator';
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/components/_all.scss:41]: Why have these been added? I had removed all components we had not worked on yet.

File: packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-buttons.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.nci-cta-button {
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-buttons.scss:1]: This should be an element, e.g. `.nci-cta-strip__button` if it is not just going to be a `usa-button` for the element.

File: packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-strip.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	ul{
+		align-items: center;
+
+		@include at-media('tablet') {
+			@include u-display("flex");
+		}
+
+		@include at-media('desktop-lg') {
+
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-strip.scss:29]: Is this meant to be empty?

File: packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-strip.scss [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		}
+
+		margin:0;
+		padding:0 units(2);
+		width: 100%;
+		li {
+			flex: 1 0 33%;
+
+			@include at-media('desktop-lg') {
+
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-strip.scss:39]: same emptyness...

File: packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-buttons.scss (hunk 2 above)
Comment (bryanpizzillo) [reply packages/ncids-css/scss/components/nci-cta-strip/internal/_cta-buttons.scss:1]: also if you are using the "primary" style as indicated with the theme token, you should be importing that style definition.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
I have a couple of things below.

Also, can you please add back in the gray boxes under the CTA strip on the inner page of the example site? No one should be approving the look and feel with it crashing on to the footer.

Finally, we really need to see this with the widescreen breakpoint working. Once we get "new dev" done for the MVP, we can make a release branch and start working on 2.0.

**Comment by arcepaul** [comment]
For people that are non-sighted or using screen readers, there is no indication that the link is external.

**Review by blilianyu — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**